pennylane/ops/qubit.py

```diff
@@ -339,21 +388,27 @@ class CSWAP(Operation):
     num_params = 0
     num_wires = 3
     par_domain = None
+    eigvals = np.array([1, 1, 1, 1, 1, 1, 1, -1])
```

Comment:
Just curious, does the ordering need to be consistent between `eigvals` and `matrix`? I would expect the last eigenvalue to be +1

Reply:
The ordering returned by `pauli_eigs` defined in `utils.py` is different in the case of Paulis (`np.array([1, -1])`) to the order outputted by e.g. `numpy` (`array([-1.,  1.])`). I'd imagine that this was followed here as well.

Reply:
This is alike to the problem I alluded to before. Basically, we can only have a concept of "order" for the eigenvalues for Observables (order needs to match with the diagonalization provided) and for DiagonalOperations (order needs to match the computational basis).

For regular gates I honestly don't know what order makes most sense. The default implementation uses numpy which spits out the eigenvalues in ascending order.

Do you have any good ideas on that?

Reply:
Eigenvalues are only used for DiagonalOperations and Observables, where the order needs to match the computational basis. For Operators falling outside this set, eigenvalues are currently not used anywhere.

In that case, I'm happy to leave the order 'undefined' for vanilla Operators, as long as we don't make any guarantee about the order in the docstring.

Reply:
I fixed this